### DISCUSSION TITLE
Is it possible to have user tasks?

### DISCUSSION THREAD

**Discussion by urumat** [PrefectHQ/prefect#6013] [Q&A]
Hello, I was wondering if it would be possible to have user tasks with Prefect. The user would have n possible replies like 'Continue', or 'Cancel' and the flow would follow different paths depending on the user reply.
Thanks in advance.

**Comment by urumat** [comment]
I would like to implement a BPM kind of flow, where a user would have 'pending tasks' that could be delegated to other users, or tasks that go to a queue where a group of users can work on. Should I start looking into other framework...?

**Reply by anna-geller** [reply]
I believe you are looking more into a user-facing application rather than building a dataflow, correct? While you could totally solve that using either parametrized flow runs or branching logic based on a value stored in a key-value pair in a JSON Block, this is certainly not a typical use case for Prefect. 

Perhaps a good question would be: what would be your end goal in how you intend to use the end solution? do you intend it so that users log into some application, do some action, and based on that, something happens in the backend? if so, this looks like you need an application rather than a dataflow.

But if you are looking for some data processing workflow where based on some input, you determine to which S3 bucket or table to load your data, or how to process an ML model, then Prefect is the right tool for you.

**Reply by urumat** [reply]
Thank you for your reply Anna.

I am thinking of a more wider approach. I would like to design data oriented flows, that is, flows with Business Intelligence functionalities.

For example: a flow that gets executed every day, first task: fetches some data from n datasources and generates a dashboard of information (sales of the day). task2: This dashboard (or Excel) is sent by email to the Manager of the company. The Manager has now a Pending Task with 2 possible replies: End flow / Go to CEO. The flow is paused now waiting for the manager reply. If the Manager responds 'Go to CEO' the dashboard is sent to the CEO and so on.

Prefect seems perfect for the data fetching/processing part but seems to be designed to some other scenarios where user interaction is not necessary.

**Reply by anna-geller** [reply]
I believe you can solve that using a flow of flows approach as shown here https://www.prefect.io/guide/blog/flow-of-flows-orchestrating-elt-with-prefect-and-dbt/
